tests/unit/h/services/annotation_stats_test.py

```diff
@@ -1,160 +1,114 @@
+        num_annotations = svc.total_user_annotation_count(sentinel.userid)
+
+        Search.assert_called_with(pyramid_request)
+        Search.return_value.clear.assert_called_once_with()
+        Limiter.assert_called_once_with()
+        DeletedFilter.assert_called_once_with()
+        UserFilter.assert_called_once_with()
+        assert Search.return_value.append_modifier.call_args_list == [
+            call(Limiter.return_value),
+            call(DeletedFilter.return_value),
+            call(UserFilter.return_value),
+        ]
+        Search.return_value.run.assert_called_with(
+            {"limit": 0, "user": sentinel.userid}
+        )
+        assert num_annotations == Search.return_value.run.return_value.total
 
-    def test_group_annotation_count_excludes_replies(
-        self, svc, search, top_level_annotation_filter
+    def test_group_annotation_count(
+        self, pyramid_request, svc, Search, TopLevelAnnotationsFilter
     ):
-        svc.group_annotation_count("groupid")
+        num_annotations = svc.group_annotation_count(sentinel.pubid)
 
-        search.return_value.append_modifier.assert_called_with(
-            top_level_annotation_filter.return_value
+        Search.assert_called_with(pyramid_request)
+        TopLevelAnnotationsFilter.assert_called_once_with()
+        Search.return_value.append_modifier.assert_called_once_with(
+            TopLevelAnnotationsFilter.return_value
         )
+        Search.return_value.run.assert_called_with(
+            {"limit": 0, "group": sentinel.pubid}
+        )
+        assert num_annotations == Search.return_value.run.return_value.total
 
-    def test_group_annotation_count_returns_total(self, svc, search):
-        search.return_value.run.return_value.total = 3
-
-        anns = svc.group_annotation_count("groupid")
-
-        assert anns == 3
+    @pytest.fixture
+    def svc(self, pyramid_request):
+        return AnnotationStatsService(request=pyramid_request)
 
 
 class TestAnnotationStatsFactory:
-    def test_returns_service(self):
-        svc = annotation_stats_factory(mock.Mock(), mock.Mock())
-
-        assert isinstance(svc, AnnotationStatsService)
-
-    def test_sets_request(self):
-        request = mock.Mock()
-        svc = annotation_stats_factory(mock.Mock(), request)
-
-        assert svc.request == request
-
-
-@pytest.fixture
-def svc(pyramid_request):
-    return AnnotationStatsService(request=pyramid_request)
+    def test_it(self, AnnotationStatsService):
+        svc = annotation_stats_factory(sentinel.context, sentinel.request)
+
+        AnnotationStatsService.assert_called_once_with(sentinel.request)
+        assert svc == AnnotationStatsService.return_value
+
+    @pytest.fixture(autouse=True)
+    def AnnotationStatsService(self, mocker):
+        return mocker.patch(
+            "h.services.annotation_stats.AnnotationStatsService",
+            autospec=True,
+            spec_set=True,
+        )
```

Comment:
Rewrote the test for the factory to use our usual style: just a single test not multiple, and mocking the service class that's being instantiated